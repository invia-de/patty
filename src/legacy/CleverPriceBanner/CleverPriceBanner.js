import React from 'react';
import ReactDOM from 'react-dom';
import PropTypes from 'prop-types';
import cx from 'classnames';
import Modal from '../../components/molecules/Modal/Modal';

import styles from './cleverpricebanner.module.scss';
import cleverPriceBanner from './img/CleverPriceBanner.png';
import cleverPriceBannerMobile from './img/CleverPriceBannerMobile.png';
import cleverPriceHeader from './img/CleverPriceHeader.png';

class CleverPriceBanner extends React.Component {
  constructor() {
    super();
    this.state = {};
  }

  render() {
    const { onClick } = this.props;

    return (
      <Modal trigger={this.renderTrigger()} onClick={onClick}>
        {this.renderModal()}
      </Modal>
    );
  }

  renderTrigger() {
    const { mobile } = this.props;
    return (
      <img
        src={mobile ? cleverPriceBannerMobile : cleverPriceBanner}
        alt="CleverPrice"
        className={cx(styles.banner, {
          [styles.mobile]: mobile,
          [styles.desktop]: !mobile
        })}
      />
    );
  }

  renderModal() {
    return (
      <div>
        <img src={cleverPriceHeader} alt="Clever Price" />
      </div>
    );
  }
}
CleverPriceBanner.propTypes = {
  /** additional classNames you want to add */
  className: PropTypes.string,
  /** required due to accessibility */
  children: PropTypes.node.isRequired,
  mobile: PropTypes.bool,
  onClick: PropTypes.func
};

export default CleverPriceBanner;

export function renderCleverPriceBanner(props, container, callback) {
  ReactDOM.render(<CleverPriceBanner {...props} />, container, callback);
}
