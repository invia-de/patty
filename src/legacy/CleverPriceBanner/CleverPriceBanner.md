CleverPriceBanner allows an `onClick` func property to be passed.
It is a cllback that is triggered, when someone clicks on the banner,
allowing GA tracking, for example.

Desktop/tablet banner:

```jsx
<div style={{ width: 267, maxWidth: '100%' }}>
  <CleverPriceBanner />
</div>
```

Mobile banner. It is automatically hidden on small devices.
Resize your window to see it.

```jsx
<CleverPriceBanner mobile />
```
