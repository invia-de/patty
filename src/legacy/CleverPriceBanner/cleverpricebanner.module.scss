@import 'theme.scss';

.banner {
  width: 100%;
  cursor: pointer;
}

// TODO: refactor to mobile-first!
.desktop {
  @include respond-below(sm) {
    display: none;
  }
}

.mobile {
  max-width: 100%;
  @include respond-above(sm) {
    display: none;
  }
}

.modal {
  font-family: 'Arial';
  max-width: 800px;
  img {
    max-width: 100%;
  }
}

.close {
  display: none;
  text-align: right;
  margin: 0 15px 15px;
  @include respond-below(sm) {
    display: block;
  }
}

.content {
  padding: 10px 20px;
  text-align: left;
  color: $blue;
  @include respond-below(sm) {
    padding: 10px 0 0;
  }
}

.header {
  img {
    width: 38px;
    float: left;
    margin: 0 5px 20px 0;
  }
  h1 {
    font-family: 'Roboto Condensed';
    font-size: 23px;
    font-weight: normal;
    line-height: 30px;
    margin: 0 0 5px;
    strong {
      font-weight: bold;
    }
  }
  h2 {
    font-family: 'Roboto Condensed';
    font-size: 18px;
    font-weight: normal;
    line-height: 22px;
    margin: 0 0 20px;
  }
  @include respond-below(sm) {
    margin: 0 15px;
  }
}

.main {
  display: flex;
  flex-direction: row-reverse;
  font-size: 12px;
  line-height: 14px;
  @include respond-below(sm) {
    flex-direction: column;
  }
}

.list {
  position: relative;
  flex-grow: 1;
  ul {
    margin: 0;
    padding: 0;
    li {
      margin: 0 0 6px 0;
      padding: 1px 0 0 25px;
      list-style: none;
      font-stretch: 80%;
      position: relative;
      &:before {
        content: '';
        position: absolute;
        left: 0;
        top: -2px;
        height: 16px;
        width: 16px;
        background-color: $blue;
        mask: url('./img/tick-2.svg') left center no-repeat;
        mask-size: 16px;
      }
    }
    @include respond-below(sm) {
      margin: 0 15px;
    }
  }
}

.links {
  display: flex;
  background-color: white;
  padding: 5px 0;
  justify-content: space-between;
  margin: 0 15px;
  @include respond-above(sm) {
    padding: 5px;
    position: absolute;
    bottom: -5px;
    left: 15px;
    width: 280px;
    margin: 0;
  }
}

.description {
  background-color: $blue-lighter;
  padding: 11px 15px 25px;
  margin-top: 5px;
  line-height: 21px;
  @include respond-below(sm) {
    padding-bottom: 11px;
  }
}

.example {
  display: flex;
  align-items: stretch;
  flex-grow: 1;
  margin-right: 10px;
  background-color: $blue-lighter;
  min-width: 257px;
  > div {
    width: 100%;
    padding: 15px;
    h1 {
      display: inline-block;
      width: 100%;
      margin: 0 0 12px;
      text-align: center;
      font-family: 'Roboto Condensed';
      font-size: 23px;
      font-weight: normal;
      line-height: 30px;
      strong {
        font-weight: bold;
      }
    }
  }
  @include respond-below(sm) {
    margin-right: 0;
    margin-top: 10px;
  }
}

.badge {
  display: inline-block;
  width: 70px;
  height: 70px;
  background-image: url('./img/CleverPriceIcon.svg');
  background-size: 47px 58px;
  background-repeat: no-repeat;
  background-position: center;
  background-color: $white;
  border-radius: 40px;
  border: 3px solid $blue;
  margin-bottom: 5px;
}

.exampleInner {
  background-color: $white;
  padding: 5px 5px 10px;
  font-size: 14px;
  line-height: 25px;
  hr {
    border-top: 3px solid $blue-lighter;
    margin: 10px 0;
  }
}

.prices {
  display: flex;
  justify-content: space-between;
}

.dots {
  flex-grow: 1;
  border-bottom: 2px dotted $blue;
  height: 16px;
}

.bigPrice {
  font-size: 20px;
}

.footer {
  display: flex;
  margin-top: 10px;
  font-size: 11px;
  line-height: 12px;
  > div:first-child {
    img {
      width: 144px;
    }
  }
  > div:nth-child(2) {
    flex-grow: 1;
    opacity: 0.7;
    padding-left: 10px;
    margin-left: 10px;
    border-left: 1px solid $blue;
  }
  @include respond-below(sm) {
    flex-direction: column;
    border-bottom: 20px solid $blue-lighter;
    > div:nth-child(1) {
      margin: 5px 15px;
    }
    > div:nth-child(2) {
      margin: 15px;
      padding-left: 0;
      border-left: 0;
    }
  }
}
