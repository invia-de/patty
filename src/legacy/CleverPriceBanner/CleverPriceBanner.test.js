import React from 'react';
import CleverPriceBanner from './CleverPriceBanner';
import { render } from '@testing-library/react';

test('Desktop CleverPriceBanner renders', () => {
  const { container } = render(<CleverPriceBanner />);
  expect(container).toMatchSnapshot();

  const trigger = container.firstChild;
  trigger.click();

  expect(container).toMatchSnapshot();
});

test('Mobile CleverPriceBanner renders', () => {
  const { container } = render(<CleverPriceBanner mobile />);
  expect(container).toMatchSnapshot();

  const trigger = container.firstChild;
  trigger.click();

  expect(container).toMatchSnapshot();
});
