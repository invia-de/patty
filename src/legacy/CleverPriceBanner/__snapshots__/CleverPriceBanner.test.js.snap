// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Desktop CleverPriceBanner renders 1`] = `
<div>
  <div
    class="trigger"
  >
    <img
      alt="CleverPrice banner"
      class="banner"
      src="CleverPriceBanner.png"
    />
  </div>
</div>
`;

exports[`Desktop CleverPriceBanner renders 2`] = `
<div>
  <div
    class="trigger"
  >
    <img
      alt="CleverPrice banner"
      class="banner"
      src="CleverPriceBanner.png"
    />
  </div>
  <div
    class="overlay overlay"
    style="opacity: 1;"
  >
    <div
      class="modal"
    >
      <div
        class="modal"
      >
        <img
          alt="CleverPrice header"
          src="CleverPriceHeader.png"
        />
        <div
          class="content"
        >
          <div
            class="close"
          >
            <button
              aria-label="Close"
              class="button"
              type="button"
            >
              <svg
                height="10px"
                id="CloseButton"
                version="1.1"
                viewBox="0 0 512 512"
                width="10px"
                x="2px"
                xmlns="http://www.w3.org/2000/svg"
                y="2px"
              >
                <path
                  d="M255.973,358.455L102.457,511.971L0,409.518l153.515-153.516L0,102.485L102.457,0.029l153.516,153.516L409.49,0.029 l102.455,102.456L358.429,256.002L512,409.518L409.49,511.971L255.973,358.455z"
                />
              </svg>
            </button>
          </div>
          <div
            class="header"
          >
            <img
              alt="CleverPrice Icon"
              src="CleverPriceIcon.svg"
            />
            <h1>
              Bei 
              <strong>
                ab-in-den-urlaub.de
              </strong>
               können Sie sich schon jetzt Lastminute-Preise sichern.
            </h1>
            <h2>
              Aus diesem Grund erhältst du, wenn du auf einer anderen Website ein billigeres Angebot findest, die Differenz von uns zurück.
            </h2>
          </div>
          <div
            class="main"
          >
            <div
              class="list"
            >
              <ul>
                <li>
                  <strong>
                    Lastminute-Preise
                  </strong>
                   schon jetzt
                   
                  <strong>
                    ab € 4,99
                  </strong>
                   sichern
                </li>
                <li>
                  bis 14 Tage vor Reisebeginn in Ruhe
                   
                  <strong>
                    bei allen Reisevermittlern
                  </strong>
                   die Preise vergleichen
                </li>
                <li>
                  <strong>
                    Entschädigung
                  </strong>
                   im Fall eines günstigeren Angebots der gleichen Reise
                </li>
                <li>
                  Erstattung von bis zu 
                  <strong>
                    10% des Reisepreises
                  </strong>
                   
                </li>
              </ul>
              <div
                class="links"
              >
                <a
                  href="https://secure.berlin-direktversicherung.de/rda-web/servlet/DokumentAdapter?dokumentTyp=PIB&uuidGroup=cab5ffa4-2aae-4363-8588-6b700a1963b1/31253"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Produktinformationen
                </a>
                <a
                  href="https://secure.berlin-direktversicherung.de/rda-web/servlet/DokumentAdapter?dokumentTyp=AVB&uuidGroup=5c02ba06-371c-444c-88ca-b951a43e91b2/31253"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Versicherungsbedingungen
                </a>
              </div>
              <div
                class="description"
              >
                Sie wollen sich Ihre Reise schon jetzt sichern und haben Angst, die Chance auf Lastminute Rabatte zu verpassen? Gönnen Sie sich die Ruhe bis zu 14 Tage vor Abreise die Preise bei uns und allen anderen Reisevermittlern zu vergleichen. Mit CleverPrice buchen Sie jetzt und erhalten bis zu 10% Ihres Reisepreises zurück, wenn Sie Ihre gebuchte Reise bei uns oder woanders günstiger finden. So sichern Sie sich jetzt schon die Chance auf Lastminute-Angebote!
                <br />
                <strong>
                  CleverPrice - Top Preise sichern!
                </strong>
              </div>
            </div>
            <div
              class="example"
            >
              <div>
                <h1>
                  <div
                    class="badge"
                  >
                     
                  </div>
                  <br />
                  <strong>
                    CleverPrice
                  </strong>
                   Beispiel
                </h1>
                <div
                  class="exampleInner"
                >
                  <div
                    class="prices"
                  >
                    <span>
                      Gekauft für
                    </span>
                    <span
                      class="dots"
                    >
                       
                    </span>
                    <span>
                      € 1998,00
                    </span>
                  </div>
                  <div
                    class="prices"
                  >
                    <span>
                      Später gefunden für
                    </span>
                    <span
                      class="dots"
                    >
                       
                    </span>
                    <span>
                      € 1799,00
                    </span>
                  </div>
                  <hr />
                  <div
                    class="prices"
                  >
                    <strong>
                      Sie erhalten
                    </strong>
                    <strong
                      class="bigPrice"
                    >
                      € 199 zurück!
                    </strong>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="footer"
          >
            <div>
              <img
                alt="Berlin Direkt"
                src="BerlinDirektLogo.png"
              />
            </div>
            <div>
              Dies ist ein Angebot der BD24 Berlin Direkt Versicherung AG vermittelt durch die Invia Travel Germany GmbH. Sämtliche Pflichtangaben im Rahmen der Versicherungsvermittlung finden Sie 
              <a
                href="https://www.ab-in-den-urlaub.de/resources/pdf/Erstinformation.pdf"
                rel="noopener noreferrer"
                target="_blank"
              >
                hier
              </a>
              .
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Mobile CleverPriceBanner renders 1`] = `
<div>
  <div
    class="trigger"
  >
    <img
      alt="CleverPrice banner"
      class="banner"
      src="CleverPriceBannerMobile.png"
    />
  </div>
</div>
`;

exports[`Mobile CleverPriceBanner renders 2`] = `
<div>
  <div
    class="trigger"
  >
    <img
      alt="CleverPrice banner"
      class="banner"
      src="CleverPriceBannerMobile.png"
    />
  </div>
  <div
    class="overlay overlay"
    style="opacity: 1;"
  >
    <div
      class="modal"
    >
      <div
        class="modal"
      >
        <img
          alt="CleverPrice header"
          src="CleverPriceHeader.png"
        />
        <div
          class="content"
        >
          <div
            class="close"
          >
            <button
              aria-label="Close"
              class="button"
              type="button"
            >
              <svg
                height="10px"
                id="CloseButton"
                version="1.1"
                viewBox="0 0 512 512"
                width="10px"
                x="2px"
                xmlns="http://www.w3.org/2000/svg"
                y="2px"
              >
                <path
                  d="M255.973,358.455L102.457,511.971L0,409.518l153.515-153.516L0,102.485L102.457,0.029l153.516,153.516L409.49,0.029 l102.455,102.456L358.429,256.002L512,409.518L409.49,511.971L255.973,358.455z"
                />
              </svg>
            </button>
          </div>
          <div
            class="header"
          >
            <img
              alt="CleverPrice Icon"
              src="CleverPriceIcon.svg"
            />
            <h1>
              Bei 
              <strong>
                ab-in-den-urlaub.de
              </strong>
               können Sie sich schon jetzt Lastminute-Preise sichern.
            </h1>
            <h2>
              Aus diesem Grund erhältst du, wenn du auf einer anderen Website ein billigeres Angebot findest, die Differenz von uns zurück.
            </h2>
          </div>
          <div
            class="main"
          >
            <div
              class="list"
            >
              <ul>
                <li>
                  <strong>
                    Lastminute-Preise
                  </strong>
                   schon jetzt
                   
                  <strong>
                    ab € 4,99
                  </strong>
                   sichern
                </li>
                <li>
                  bis 14 Tage vor Reisebeginn in Ruhe
                   
                  <strong>
                    bei allen Reisevermittlern
                  </strong>
                   die Preise vergleichen
                </li>
                <li>
                  <strong>
                    Entschädigung
                  </strong>
                   im Fall eines günstigeren Angebots der gleichen Reise
                </li>
                <li>
                  Erstattung von bis zu 
                  <strong>
                    10% des Reisepreises
                  </strong>
                   
                </li>
              </ul>
              <div
                class="links"
              >
                <a
                  href="https://secure.berlin-direktversicherung.de/rda-web/servlet/DokumentAdapter?dokumentTyp=PIB&uuidGroup=cab5ffa4-2aae-4363-8588-6b700a1963b1/31253"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Produktinformationen
                </a>
                <a
                  href="https://secure.berlin-direktversicherung.de/rda-web/servlet/DokumentAdapter?dokumentTyp=AVB&uuidGroup=5c02ba06-371c-444c-88ca-b951a43e91b2/31253"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Versicherungsbedingungen
                </a>
              </div>
              <div
                class="description"
              >
                Sie wollen sich Ihre Reise schon jetzt sichern und haben Angst, die Chance auf Lastminute Rabatte zu verpassen? Gönnen Sie sich die Ruhe bis zu 14 Tage vor Abreise die Preise bei uns und allen anderen Reisevermittlern zu vergleichen. Mit CleverPrice buchen Sie jetzt und erhalten bis zu 10% Ihres Reisepreises zurück, wenn Sie Ihre gebuchte Reise bei uns oder woanders günstiger finden. So sichern Sie sich jetzt schon die Chance auf Lastminute-Angebote!
                <br />
                <strong>
                  CleverPrice - Top Preise sichern!
                </strong>
              </div>
            </div>
            <div
              class="example"
            >
              <div>
                <h1>
                  <div
                    class="badge"
                  >
                     
                  </div>
                  <br />
                  <strong>
                    CleverPrice
                  </strong>
                   Beispiel
                </h1>
                <div
                  class="exampleInner"
                >
                  <div
                    class="prices"
                  >
                    <span>
                      Gekauft für
                    </span>
                    <span
                      class="dots"
                    >
                       
                    </span>
                    <span>
                      € 1998,00
                    </span>
                  </div>
                  <div
                    class="prices"
                  >
                    <span>
                      Später gefunden für
                    </span>
                    <span
                      class="dots"
                    >
                       
                    </span>
                    <span>
                      € 1799,00
                    </span>
                  </div>
                  <hr />
                  <div
                    class="prices"
                  >
                    <strong>
                      Sie erhalten
                    </strong>
                    <strong
                      class="bigPrice"
                    >
                      € 199 zurück!
                    </strong>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="footer"
          >
            <div>
              <img
                alt="Berlin Direkt"
                src="BerlinDirektLogo.png"
              />
            </div>
            <div>
              Dies ist ein Angebot der BD24 Berlin Direkt Versicherung AG vermittelt durch die Invia Travel Germany GmbH. Sämtliche Pflichtangaben im Rahmen der Versicherungsvermittlung finden Sie 
              <a
                href="https://www.ab-in-den-urlaub.de/resources/pdf/Erstinformation.pdf"
                rel="noopener noreferrer"
                target="_blank"
              >
                hier
              </a>
              .
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
